lib/project_types/script/layers/infrastructure/assemblyscript_task_runner.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class AssemblyScriptTaskRunner
+        BYTECODE_FILE = "%{name}.wasm"
+        SCRIPT_SDK_BUILD = "npx --no-install shopify-scripts-build --src=../%{source} --binary=#{BYTECODE_FILE} "\
+                           "-- --lib=../node_modules --validate --optimize"
+
+        attr_reader :ctx
+
+        def initialize(ctx)
+          @ctx = ctx
+        end
+
+        def build
+          compile
+          bytecode
+        end
+
+        def compiled_type
+          "wasm"
+        end
+
+        def install_dependencies
+          check_node_version!
+
+          output, status = ctx.capture2e("npm", "install", "--no-audit", "--no-optional", "--loglevel error")
+          raise Errors::DependencyInstallError, output unless status.success?
+        end
+
+        def dependencies_installed?
+          # Assuming if node_modules folder exist at root of script folder, all deps are installed
+          Dir.exist?("node_modules")
+        end
+
+        private
+
+        def project
+          @project ||= ScriptProject.current
+        end
```

Comment:
I thin we can avoid the dependency on `ScriptProject` here in this TaskRunner.  The current dependency means that, any time `ScriptProject` changes (imagine the `.current` API changes), this class needs to change, which seems unnecessary.  We could just pass in the `project.source_file` and `project.script_name` either in the `initialize` or `compile`.  WDYT?

Reply:
Thats a good point. The only issue I have with that is depending on the language, there may be multiple sources and inputs which would complicate the common task runner interface.

I think ideally, we would need to have classes like `AssemblyScriptProject` and `KotlinProject` or something which would then instantiate both their respective task runner and project creator (ie access the task runner with `assemblyscript_project.task_runner`). Then, that project instance could just inject itself (or the source files and other fields) as a parameter to the TaskRunner. This way, we can avoid calling the `.current` method here and still have a clean interface for task runners that need different inputs.

Reply:
Talked on slack - We'll just pass in the `source_file` and `script_name` for now. Once we actually need to support other languages then we'll come back to it!